### PR TITLE
feat(obj): add lifecycle to object-storage

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: docs/resources/object_bucket.md [hunk 1]
```diff
@@ -21,6 +21,85 @@ resource "scaleway_object_bucket" "some_bucket" {
...
+        days = 365
+      }
+  
+      transition {
+        days          = 120
+        storage_class = "GLACIER"
+      }
+  }
+  
+  lifecycle_rule {
```
Comment (remyleone) [inline docs/resources/object_bucket.md:47]: For all those lifecycle_rule, could you add a description in English that explains what they do?

File: docs/resources/object_bucket.md [hunk 2]
```diff
@@ -40,6 +119,27 @@ The `CORS` object supports the following:
 * `expose_headers` (Optional) Specifies expose header in the response.
 * `max_age_seconds` (Optional) Specifies time in seconds that browser can cache the response for a preflight request.
 
+The `lifecycle_rule` (Optional) object supports the following:
+
+* `id` - (Optional) Unique identifier for the rule. Must be less than or equal to 255 characters in length.
```
Comment (remyleone) [inline docs/resources/object_bucket.md:124]: Could you add an example without id? Because it is present everytime it gives the impression that it is mandatory.

File: docs/resources/object_bucket.md [hunk 3]
```diff
@@ -21,6 +21,85 @@ resource "scaleway_object_bucket" "some_bucket" {
...
+        "terraform" = "hashicorp"
+      }
+  
+      transition {
+        days          = 0
+        storage_class = "GLACIER"
+      }
+  }
+  
+  lifecycle_rule {
```
Comment (remyleone) [inline docs/resources/object_bucket.md:87]: What is the difference between this example and the one above? Is it really necessary?

File: docs/resources/object_bucket.md [hunk 4]
```diff
@@ -40,6 +119,27 @@ The `CORS` object supports the following:
...
+* `enabled` - (Required) Specifies lifecycle rule status.
+* `abort_incomplete_multipart_upload_days` (Optional) Specifies the number of days after initiating a multipart upload when the multipart upload must be completed.
+* `expiration` - (Optional) Specifies a period in the object's expire (documented below).
+* `transition` - (Optional) Specifies a period in the object's transitions (documented below).
+
+At least one of `abort_incomplete_multipart_upload_days`, `expiration`, `transition` must be specified.
+
+The `expiration` object supports the following
+
+* `days` (Optional) Specifies the number of days after object creation when the specific rule action takes effect.
```
Comment (remyleone) [inline docs/resources/object_bucket.md:136]: Does it support float if I want something less than a day? Could you add an example?

File: docs/resources/object_bucket.md [hunk 5]
```diff
@@ -40,6 +119,27 @@ The `CORS` object supports the following:
...
+
+The `transition` object supports the following
+
+* `days` (Optional) Specifies the number of days after object creation when the specific rule action takes effect.
+* `storage_class` (Required) Specifies the Scaleway storage class `STANDARD`, `GLACIER`, `ONEZONE_IA`  to which you want the object to transition.
```
Comment (remyleone) [inline docs/resources/object_bucket.md:141]: Could you add a link to https://www.scaleway.com/en/docs/storage/object/concepts/#storage-class

File: scaleway/resource_object_bucket.go [hunk 6]
```diff
@@ -88,6 +92,73 @@ func resourceScalewayObjectBucket() *schema.Resource {
 					},
 				},
 			},
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:         schema.TypeString,
```
Comment (remyleone) [inline scaleway/resource_object_bucket.go:101]: Some descriptions are missing

File: docs/resources/object_bucket.md (hunk 4 above)
Comment (Monitob) [reply docs/resources/object_bucket.md:136]: Yes, If you specify the days’ value as 0, the objects are eligible to be sent to Glacier at midnight UTC following its creation.

File: docs/resources/object_bucket.md (hunk 2 above)
Comment (Monitob) [reply docs/resources/object_bucket.md:122]: I've just added a new  test and example in the documentation

File: docs/resources/object_bucket.md (hunk 3 above)
Comment (Monitob) [reply docs/resources/object_bucket.md:93]: I've modify the use case

File: docs/resources/object_bucket.md [hunk 7]
```diff
@@ -21,6 +21,83 @@ resource "scaleway_object_bucket" "some_bucket" {
...
 
+### Using object lifecycle
+
+```hcl
+resource "scaleway_object_bucket" "main"{
+  name = "mybuckectid"
+  region = "fr-par"
+  acl = "private"
+  
+  # Rule case with storage classes GLACIER designed for low-cost data archiving
```
Comment (remyleone) [inline docs/resources/object_bucket.md:32]: Could you do a sentence in english that describe what this lifecycle does. For instance:

This rule will make all object that got a key that start with `path1/` be transferred from their default storage class (`STANDARD`, `ONEZONE_IA`) to GLACIER after 120 days counting from their creation and then 365 days after that they will be expired and deleted.

File: docs/resources/object_bucket.md [hunk 8]
```diff
@@ -21,6 +21,89 @@ resource "scaleway_object_bucket" "some_bucket" {
...
+        days          = 0
+        storage_class = "GLACIER"
+      }
+  }
+ 
+  # This lifecycle configuration rule specifies with the AbortIncompleteMultipartUpload action to 
+  # stop incomplete multipart uploads (identified by the key name prefix (path5/) in the rule)
+  # if they aren't completed within a specified number of days after initiation.
+  lifecycle_rule {
+      prefix  = "path5/"
```
[the remark this answers is not in the document]
Comment (Monitob) [reply docs/resources/object_bucket.md]: Thanks for the suggestion. I've added this to the documentation. It's highly recommended to not set a prefix with this kind of configuration.

File: docs/resources/object_bucket.md [hunk 9]
```diff
@@ -40,6 +123,27 @@ The `CORS` object supports the following:
...
+At least one of `abort_incomplete_multipart_upload_days`, `expiration`, `transition` must be specified.
+
+The `expiration` object supports the following
+
+* `days` (Optional) Specifies the number of days after object creation when the specific rule action takes effect.
+
+The `transition` object supports the following
+
+* `days` (Optional) Specifies the number of days after object creation when the specific rule action takes effect.
+* `storage_class` (Required) Specifies the Scaleway [storage class](https://www.scaleway.com/en/docs/storage/object/concepts/#storage-class) `STANDARD`, `GLACIER`, `ONEZONE_IA`  to which you want the object to transition.
```
[the remark this answers is not in the document]
Comment (Monitob) [reply docs/resources/object_bucket.md:149]: I've put an important section for this. Thanks

File: docs/resources/object_bucket.md [hunk 10]
```diff
@@ -21,6 +21,89 @@ resource "scaleway_object_bucket" "some_bucket" {
...
+      prefix  = "path2/"
+      enabled = true
+  
+      expiration {
+        days = "50"
+      }
+  }
+  
+  # This lifecycle configuration rule directs Scaleway S3 to transition objects with the (path3/)
+  # prefix to the S3 Glacier Flexible Retrieval storage class soon after creation.
```
[the remark this answers is not in the document]
Comment (Monitob) [reply docs/resources/object_bucket.md]: Ok, I've updated the description. Thanks for the suggestion.

File: scaleway/resource_object_bucket.go [hunk 11]
```diff
@@ -88,18 +92,97 @@ func resourceScalewayObjectBucket() *schema.Resource {
...
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional:    true,
+							Description: "The tags associated with the bucket lifecycle",
+						},
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Specifies the expiration status",
```
[the remark this answers is not in the document]
Comment (Monitob) [reply scaleway/resource_object_bucket.go]: The element value can be either Enabled or Disabled. I've just added a new description

File: scaleway/resource_object_bucket.go [hunk 12]
```diff
@@ -88,18 +92,97 @@ func resourceScalewayObjectBucket() *schema.Resource {
...
 			"versioning": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Computed: true,
-				MaxItems: 1,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Description: "keeping multiple variants of an object in the same bucket",
```
Comment (remyleone) [inline scaleway/resource_object_bucket.go:178]:
```suggestion
				Description: "Allow multiple versions of an object in the same bucket",
```

File: scaleway/provider_test.go [hunk 13]
```diff
@@ -67,6 +67,7 @@ func getHTTPRecoder(t *testing.T, update bool) (client *http.Client, cleanup fun
 		i.Request.Headers = i.Request.Headers.Clone()
 		delete(i.Request.Headers, "x-auth-token")
 		delete(i.Request.Headers, "X-Auth-Token")
+		delete(i.Request.Headers, "Authorization")
```
Comment (remyleone) [inline scaleway/provider_test.go:70]: Why is this change required? I thought is was needed by #1078

File: scaleway/testdata/iot-route-minimal.cassette.yaml [hunk 14]
```diff
@@ -8,12 +8,12 @@ interactions:
       Content-Type:
```
Comment (remyleone) [inline scaleway/testdata/iot-route-minimal.cassette.yaml]: Should not be modified in this PR.

File: scaleway/provider_test.go (hunk 13 above)
Comment (Monitob) [reply scaleway/provider_test.go]: Yes, I've just removed from here

File: scaleway/resource_object_bucket_test.go [hunk 15]
```diff
@@ -103,6 +105,223 @@ func TestAccScalewayObjectBucket_Basic(t *testing.T) {
...
+							}
+
+							transition {
+							  days          = 30
+							  storage_class = "STANDARD"
+							}
+						}
+					}
+				`, bucketLifecycle),
+				Check: resource.ComposeTestCheckFunc(
```
[the remark this answers is not in the document]
Comment (Monitob) [reply scaleway/resource_object_bucket_test.go:132]: I just added lifecycle verification test for this.

File: scaleway/testdata/iot-route-minimal.cassette.yaml (hunk 14 above)
Comment (Monitob) [reply scaleway/testdata/iot-route-minimal.cassette.yaml:8]: done

**Review by remyleone — APPROVED** [review]
